### PR TITLE
(GAP-1984) Adds Navbar for authenticated users

### PR DESCRIPTION
**Pull request by john-tco**
## Description

[GAP-1984](https://technologyprogramme.atlassian.net/jira/software/projects/GAP/boards/216?selectedIssue=GAP-1984)

todo - drill the `isUserLoggedIn` prop into every single page OR remove Layout from every single page and render `Layout` from app entrypoint

ui

Screenshot 2023-11-07 at 09 45 51
Screenshot 2023-11-07 at 09 46 15

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

[GAP-1984]: https://technologyprogramme.atlassian.net/browse/GAP-1984?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: __tests__/pages/notifications/deleteSearch.test.js [hunk 1]
```diff
@@ -21,15 +21,17 @@ jest.mock('../../../src/service/api-key-service');
 jest.mock('../../../src/utils/contentFulPage');
 jest.mock('../../../src/utils/cookieAndJwtChecker');
 jest.mock('../../../src/service/saved_search_service');
+jest.mock('next/config', () => () => ({
+  publicRuntimeConfig: {},
+}));
```
Comment (dominicwest) [inline __tests__/pages/notifications/deleteSearch.test.js:26]: If this is repeated a lot, can setup jest to run it before each test using some config

File: pages/api/logout/index.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+const Logout = async (req: NextApiRequest, res: NextApiResponse) => {
+  const sessionCookie = getSessionIdFromCookies(req);
+  if (sessionCookie) await logoutAdmin(sessionCookie);
+
+  res.setHeader(
+    'Set-Cookie',
+    `session_id=deleted; Path=/; secure; HttpOnly; SameSite=Strict; expires=Thu, 01 Jan 2003 00:00:00 GMT`,
+  );
+  const logoutUrl =
+    process.env.ONE_LOGIN_ENABLED === 'true'
```
Comment (dylanwrightCO) [inline pages/api/logout/index.ts:14]: I'm not sure there is a scenario where the user can logout if One Login is disabled.  In that case we shouldn't display the logout link

Comment (john-tco) [reply pages/api/logout/index.ts:14]: Admins can still logout from their admin session from find. We could render it for admins only?

**Review by dylanwrightCO — DISMISSED** [review]

File: src/utils/session.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { NextApiRequest } from 'next';
+import { NextApiRequestCookies } from 'next/dist/server/api-utils';
+import { NextIncomingMessage } from 'next/dist/server/request-meta';
+
+const getSessionIdFromCookies = (
+  req:
+    | (NextIncomingMessage & { cookies: NextApiRequestCookies })
+    | NextApiRequest,
+) => {
+  if (!process.env.SESSION_COOKIE_NAME) {
```
Comment (jgunnCO) [inline src/utils/session.ts:10]: should add `SESSION_COOKIE_NAME` to `.env.example`

File: pages/api/logout/index.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import { getSessionIdFromCookies } from '../../../src/utils/session';
+import axios from 'axios';
+
+const Logout = async (req: NextApiRequest, res: NextApiResponse) => {
+  const sessionCookie = getSessionIdFromCookies(req);
+  if (sessionCookie) await logoutAdmin(sessionCookie);
+
+  res.setHeader(
+    'Set-Cookie',
+    `session_id=deleted; Path=/; secure; HttpOnly; SameSite=Strict; expires=Thu, 01 Jan 2003 00:00:00 GMT`,
```
Comment (jgunnCO) [inline pages/api/logout/index.ts:11]: should this be `process.env.USER_TOKEN_NAME`?

Comment (jgunnCO) [reply pages/api/logout/index.ts:11]:
```suggestion
    `${process.env.USER_TOKEN_NAME}=deleted; Path=/; secure; HttpOnly; SameSite=Strict; expires=Thu, 01 Jan 2003 00:00:00 GMT`,
```

File: pages/api/logout/index.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  headers: {
+    Cookie: `SESSION=${sessionId};`,
+  },
+});
+
+const logoutAdmin = async (sessionCookie: string) =>
+  axios.delete(
+    `${process.env.ADMIN_BACKEND_HOST}/logout`,
+    axiosSessionConfig(sessionCookie),
+  );
```
Comment (jgunnCO) [inline pages/api/logout/index.ts:27]: and then if you need to delete `session_id` (which I think you've got as `process.env.SESSION_COOKIE_NAME`?) do it here?

File: pages/api/logout/index.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+const Logout = async (req: NextApiRequest, res: NextApiResponse) => {
+  const sessionCookie = getSessionIdFromCookies(req);
+  if (sessionCookie) await logoutAdmin(sessionCookie);
+
+  res.setHeader(
+    'Set-Cookie',
+    `session_id=deleted; Path=/; secure; HttpOnly; SameSite=Strict; expires=Thu, 01 Jan 2003 00:00:00 GMT`,
+  );
+  res.redirect(302, process.env.V2_LOGOUT_URL);
```
Comment (jgunnCO) [inline pages/api/logout/index.ts:13]: need to add this to `.env.example` as well

File: pages/api/logout/index.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+const axiosSessionConfig = (sessionId: string) => ({
+  withCredentials: true,
+  headers: {
+    Cookie: `SESSION=${sessionId};`,
+  },
+});
+
+const logoutAdmin = async (sessionCookie: string) =>
+  axios.delete(
+    `${process.env.ADMIN_BACKEND_HOST}/logout`,
```
Comment (jgunnCO) [inline pages/api/logout/index.ts:25]: also needs to go in `.env.example`

Comment (jgunnCO) [inline pages/api/logout/index.ts:25]: also needs to go in `.env.example`

Comment (john-tco) [reply pages/api/logout/index.ts:25]: all fair

**Review by jgunnCO — APPROVED** [review]

**Pull request merged**